internal/expr/scan.go

```diff
@@ -0,0 +1,196 @@
+package expr
+
+import (
+	"database/sql"
+	"fmt"
+	"reflect"
+	"strconv"
+)
+
+// ResultExpr represents the result of a database query.
+type ResultExpr struct {
+	outputs []field
+	rows    *sql.Rows
```

Comment:
Per our last call, the API for this type is what the spec defines as `Query`. This is breaking away the notion that this package is about expresssions (`expr`), and introducing concepts which are about handling the database conenction and interactions.

Is this what we want?  What alternatives were considered?

Reply:
That's a good point, I think this scanning side of it would be better suited in a separate package.